pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -71,6 +72,14 @@ const (
 )
 
 const (
+	cephOSDStart = `%s %s
+rc=$?
+if [ $rc -eq 0 ]; then
+	echo "OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6 hrs. Restart the pod manually once the flapping issue is fixed"
+	sleep 21600s
```

Comment:
@subhamkrai Although the script is reading a 0 exit code, but its not sleeping for 6 hrs. The pod restarts after around 10 seconds. Am I missing anything in the script? 

OSD pod logs after 0 exit code:

``` debug2023-08-14T05:25:00.471+0000 7f7179658700 -1 osd.0 34 _committed_osd_maps marked down 6 > osd_max_markdown_count 5 in last 600.000000 seconds, shutting down
debug2023-08-14T05:25:00.471+0000 7f7179658700  1 osd.0 34 start_waiting_for_healthy
debug2023-08-14T05:25:00.476+0000 7f7179658700  0 osd.0 34 _committed_osd_maps shutdown OSD via async signal
debug2023-08-14T05:25:00.476+0000 7f7188871700 -1 received  signal: Interrupt from Kernel ( Could be generated by pthread_kill(), raise(), abort(), alarm() ) UID: 0
debug2023-08-14T05:25:00.476+0000 7f7188871700 -1 osd.0 34 *** Got signal Interrupt ***
debug2023-08-14T05:25:00.476+0000 7f7188871700  0 osd.0 34 Fast Shutdown: - cct->_conf->osd_fast_shutdown = 1, null-fm = 0
debug2023-08-14T05:25:00.476+0000 7f7188871700 -1 osd.0 34 *** Immediate shutdown (osd_fast_shutdown=true) ***
debug2023-08-14T05:25:00.476+0000 7f7188871700  0 osd.0 34 prepare_to_stop starting shutdown
OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 6 hrs. Restart the pod manually once the flapping issue is fixed
```

Reply:
maybe try h for hour

Reply:
tried 6h. Still same. 

Tried 15m. No luck.

``` debug2023-08-14T06:29:23.558+0000 7fad1aacc700  0 log_channel(cluster) log [DBG] : map e33 wrongly marked me down at e33
debug2023-08-14T06:29:23.558+0000 7fad1aacc700 -1 osd.0 33 _committed_osd_maps marked down 6 > osd_max_markdown_count 5 in last 600.000000 seconds, shutting down
debug2023-08-14T06:29:23.558+0000 7fad1aacc700  1 osd.0 33 start_waiting_for_healthy
debug2023-08-14T06:29:23.559+0000 7fad1aacc700  0 osd.0 33 _committed_osd_maps shutdown OSD via async signal
debug2023-08-14T06:29:23.561+0000 7fad29ce5700 -1 received  signal: Interrupt from Kernel ( Could be generated by pthread_kill(), raise(), abort(), alarm() ) UID: 0
debug2023-08-14T06:29:23.561+0000 7fad29ce5700 -1 osd.0 33 *** Got signal Interrupt ***
debug2023-08-14T06:29:23.561+0000 7fad29ce5700  0 osd.0 33 Fast Shutdown: - cct->_conf->osd_fast_shutdown = 1, null-fm = 0
debug2023-08-14T06:29:23.561+0000 7fad29ce5700 -1 osd.0 33 *** Immediate shutdown (osd_fast_shutdown=true) ***
debug2023-08-14T06:29:23.561+0000 7fad29ce5700  0 osd.0 33 prepare_to_stop starting shutdown
+ rc=0
OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 15 minutes. Restart the pod manually once the flapping issue is fixed
+ '[' 0 -eq 0 ']'
+ echo 'OSD daemon exited with code 0, possibly due to OSD flapping. The OSD pod will sleep for 15 minutes. Restart the pod manually once the flapping issue is fixed'
+ sleep 15m
```

Reply:
As discussed, try disabling the liveness probe. It's probably causing the pod to be restarted since the probe wouldn't respond with the osd not running anymore.

Reply:
That looks like the reason.

Reply:
@sp98 can we add the same script in livenessProbe to sleep for 6h if exit code of the command in zero of osd daemons?

Reply:
The livenessProbe needs to complete quickly, otherwise the kubelet will think the probe failed and the pod needs to be restarted. 
So the question is how the liveness probe can detect that the osd is sleeping intentionally, and then not fail the probe in that case. Perhaps the probe could execute a command similar to `ps aux | grep sleep` to see if the osd is sleeping.

Reply:
Tried creating a tmp file if OSD fails with exit code 0.  Changed livenessprobe to not exit if the tmp file is available.